multithreading update (PR #42340) leads to build error on macOS arm64 machine

PR #42340 led to this Julia master build error on macOS with Apple Silicon:

```
(base) gong@polaris julia % make                   
Creating usr/etc/julia/startup.jl
Copying in usr/share/man/man1/julia.1
/Users/gong/GitHub/julia/contrib/install.sh 755 /Users/gong/GitHub/julia/contrib/julia-config.jl /Users/gong/GitHub/julia/usr/share/julia/
    PERL base/pcre_h.jl
    PERL base/uv_constants.jl
    LINK usr/lib/libjulia.1.9.dylib
    LINK usr/lib/libjulia.1.dylib
    LINK usr/lib/libjulia.dylib
    LINK usr/bin/julia
    CC src/jloptions.o
/Users/gong/GitHub/julia/src/jloptions.c:449:37: error: implicit declaration of function 'jl_effective_threads' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                jl_options.nprocs = jl_effective_threads();
                                    ^
1 error generated.
make[1]: *** [jloptions.o] Error 1
make: *** [julia-src-release] Error 2
```

--- a/src/jloptions.c
+++ b/src/jloptions.c
@@ -4,6 +4,7 @@
 #include <errno.h>
 
 #include "julia.h"
+#include "julia_internal.h"
 
 #include <unistd.h>
 #include <getopt.h>

